tests/signal_manager_test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+. ./src/signal_manager.sh --source-only
+. ./tests/utils --source-only
+
+function suite() {
+  suite_addTest "default_handler_Test"
+  suite_addTest "non_signal_Test"
+  suite_addTest "new_signal_Test"
+  suite_addTest "signal_reset_Test"
+  suite_addTest "add_two_signals_Test"
+}
+
+function default_handler_Test() {
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'"Oh no! An interruption! See ya..."
+  local output
+
+  cd "$SHUNIT_TMPDIR"
```

Comment:
What is this `$SHUNIT_TMPDIR`? That's something new to me; it looks interesting.

Reply:
It's useful! See [shunit's readme](https://github.com/kward/shunit2/#-advanced-usage): 

> SHUNIT_TMPDIR | Path to temporary directory that will be automatically cleaned up upon exit of shUnit2.